### DISCUSSION TITLE
Image Resizing - Magic Kernel vs. AVIR - AVIR vs. Magic Kernel?

### DISCUSSION THREAD

**Discussion by ghost** [ImageMagick/ImageMagick#3316] [Digital Image Processing] [answered]
### Magic Kernel

http://johncostella.com/magic/

### AVIR

https://github.com/avaneev/avir

Would you consider adding both of these image resizing methods to ImageMagick?

**Comment by urban-warrior** [comment, marked as answer]
Have you reviewed https://legacy.imagemagick.org/Usage/resize/ and https://legacy.imagemagick.org/Usage/filter/nicolas/?  If you can quantify how either of the proposed resizing methods improve upon the plethora of existing ImageMagick resize methods (and there are many), we will consider taking a closer look at your proposed algorithms.

**Reply by ghost** [reply]
Yes and yes. Great information via those two links.

For me to present additional data, other than the already extensive and well documented papers, from both the Magic Kernel and the AVIR developers, would be fruitless. The above links should provide plenty of facts. Do take a closer look at the proposed algorithms.

**Reply by urban-warrior** [reply]
Thanks for the tip but unless someone can prove the algorithms add value that's not already captured by the resizing methods already contained within ImageMagick, we are not inclined to invest any time into this effort.  However, ImageMagick is open-source.  Any user can contribute these algorithms as a pull request @ https://github.com/ImageMagick/ImageMagick/pulls.

**Reply by ghost** [reply]
Here's Magic Kernel's algorithm proof:

http://johncostella.com/magic/mks.pdf

Thanks for the tip!

**Reply by urban-warrior** [reply]
We already read that paper.  It discusses why the resizing algorithm has some nice properties for resizing with sharpening.  ImageMagick already has multiple options for resizing with sharpening that is comparable to the results we see in the paper.  Again, we are not interested in investing time unless an algorithm has provable results that are not already currently addressed in ImageMagick.  The ImageMagick developers have thousands of issues to address so we need to prioritize our time/effort. For example, today we're implementing reference illuminant tristimulus values (e.g. D65, D50, etc.), something that was requested in 2018. However, you or another user can contribute the algorithm as a pull request as we discussed in our previous message.

**Reply by ghost** [reply]
Understood.

Could the new reference illuminant tristimulus values, be used to change embedded and/or non-embedded ICC profile information, could they be used to change non-ICC profile color space information?

**Reply by urban-warrior** [reply]
We like to keep each discussion atomic, one topic per.  Feel free to create a separate discussion and we'll answer your questions/concerns there.

**Reply by ghost** [reply]
Ok.
